src/classes/ADDR_Seasonal_UTIL.cls

```diff
@@ -0,0 +1,261 @@
+        integer startYear = Integer.valueOf(addr.Seasonal_Start_Year__c);
+        integer startMonth = Integer.valueOf(addr.Seasonal_Start_Month__c);
+        integer startDay = Integer.valueOf(addr.Seasonal_Start_Day__c);
+        
+        integer endYear = Integer.valueOf(addr.Seasonal_End_Year__c);
+        integer endMonth = Integer.valueOf(addr.Seasonal_End_Month__c);
+        integer endDay = Integer.valueOf(addr.Seasonal_End_Day__c);
+        
+        Date startDate;
+        Date endDate;
+        Date dtToday = System.today();
+        
+        //start and end year have been provided (we don't allow providing only start or only end - addrSeasonalPartialYear error)
+        if(startYear != null && endYear != null) {
+            startDate = Date.newInstance(startYear, startMonth, startDay);
+            endDate = Date.newInstance(endYear, endMonth, endDay);
+        //start and end year have not been provided
+        } else {
+            //seasonal address is contained within a single year
+            if (startMonth < endMonth ||(startMonth == endMonth && startDay <= endDay)) {
+                UTIL_Debug.debug('****Seasonal address contained within a single year');
+                startDate = Date.newInstance(dtToday.year(), startMonth, startDay);
+                endDate = Date.newInstance(dtToday.year(), endMonth, endDay);
+            //seasonal address overlaps the year, and current month is past the start month
+            } else if (startMonth <= dtToday.month()){
+                UTIL_Debug.debug('****seasonal address overlaps the year, and current month is past the start month');
+                startDate = Date.newInstance(dtToday.year(), startMonth, startDay);
+                endDate = Date.newInstance(dtToday.year()+1, endMonth, endDay);
+            //seasonal address overlaps the year, and current month is before the start month
+            } else {
+                UTIL_Debug.debug('****seasonal address overlaps the year, and current month is before the start month');
+                startDate = Date.newInstance(dtToday.year()-1, startMonth, startDay);
+                endDate = Date.newInstance(dtToday.year(), endMonth, endDay);
+            }
+        }
+        //to return the start or end date, depending on the second parameter passed to the method
+        return (returnStartDate ? startDate : endDate);
+    }
+    
+    /*******************************************************************************************************
+    * @description utility to see if the address has seasonal fields set
+    * @param addr the Address to look at
+    * @return Boolean
+    */ 
+    public static Boolean isSeasonalAddr(Address__c addr) {
+        return (addr.Seasonal_Start_Month__c != null && addr.Seasonal_Start_Day__c != null &&
+           addr.Seasonal_End_Month__c != null && addr.Seasonal_End_Day__c != null);
+    }
```

Comment:
We have start and end year, but they are optional. If those don't have a value we assume is a recurring address.